etna/transforms/math/limit.py

```diff
@@ -0,0 +1,135 @@
+        ------
+        ValueError:
+            Some ``in_column`` features are less than ``lower_bound`` or greater than ``upper_bound``.
+        """
+        super().__init__(required_features=[in_column])
+        self.in_column = in_column
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+        self.tol = tol
+
+    def _fit(self, df: pd.DataFrame):
+        """Fit method does nothing and is kept for compatibility.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data.
+        """
+        pass
+
+    def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Apply scaled logit transform to the dataset.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data to transform.
+
+        Returns
+        -------
+        :
+            transformed dataframe
+
+        """
+        # TODO: fix precision
+        if self.lower_bound is None and self.upper_bound is None:
+            transformed_features = df
+        elif self.lower_bound is not None and self.upper_bound is None:
+            transformed_features = np.log((df - self.lower_bound + self.tol))
+        elif self.lower_bound is None and self.upper_bound is not None:
+            transformed_features = np.log((self.upper_bound + self.tol - df))
+        else:
+            transformed_features = np.log((df - self.lower_bound + self.tol) / (self.upper_bound + self.tol - df))  # type: ignore
+
+        if (self.lower_bound is not None and (df < self.lower_bound).any().any()) or (
+            self.upper_bound is not None and (df > self.upper_bound).any().any()
+        ):
+            left_border = self.lower_bound
+            right_border = self.upper_bound
+            if self.lower_bound is None:
+                left_border = np.NINF
+            if self.upper_bound is None:
+                right_border = np.inf
+            raise ValueError(f"Detected values out [{left_border}, {right_border}]")
+        return transformed_features
+
+    def _inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Apply scaled logit reverse transform to the dataset.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data to transform.
+
+        Returns
+        -------
+        :
+            transformed series
+        """
+        # TODO: fix precision
+        if self.lower_bound is None and self.upper_bound is None:
+            transformed_features = df
+        elif self.lower_bound is not None and self.upper_bound is None:
+            transformed_features = np.exp(df) + self.lower_bound - self.tol
+        elif self.lower_bound is None and self.upper_bound is not None:
+            transformed_features = self.upper_bound - np.exp(df) + self.tol
+        else:
+            exp_df = np.exp(df)
+            transformed_features = ((self.upper_bound - self.lower_bound) * exp_df) / (1 + exp_df) + self.lower_bound  # type: ignore
```

Comment:
mypy marked that "Unsupported operand types for + ("None" and "float")" and I don't know how to fix it.
my code logic is good

Reply:
We should investigate the problem deeper, I will look at it

Reply:
Looks like it is hard to mypy to derive the typing in this case, if you write
```
elif self.lower_bound is not None and self.upper_bound is not None:
```
it will be ok. Let's leave the ignore directive